docs/shared-mutable-state-and-concurrency.md

```diff
@@ -73,30 +74,33 @@ multi-threaded [Dispatchers.Default] that is used in [GlobalScope].
 import kotlinx.coroutines.*
 import kotlin.system.*    
 
-suspend fun CoroutineScope.massiveRun(action: suspend () -> Unit) {
+suspend fun massiveRun(action: suspend () -> Unit) {
     val n = 100  // number of coroutines to launch
     val k = 1000 // times an action is repeated by each coroutine
     val time = measureTimeMillis {
-        val jobs = List(n) {
-            launch {
-                repeat(k) { action() }
+        coroutineScope { // scope for coroutines 
+            repeat(n) {
+                launch {
+                    repeat(k) { action() }
+                }
             }
         }
-        jobs.forEach { it.join() }
     }
     println("Completed ${n * k} actions in $time ms")    
 }
 
+//sampleStart
 var counter = 0
 
-fun main() = runBlocking<Unit> {
-//sampleStart
-    GlobalScope.massiveRun {
-        counter++
+fun main() = runBlocking {
+    withContext(Dispatchers.Default) {
```

Comment:
Isn't this effectively the same as `suspend fun main()`?

Reply:
Yes. But it looks more consistent across various examples in this chapter when it is written this way.